packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/util/SchemaUtil.java

```diff
@@ -276,11 +277,11 @@ public static ClassType mapExternalModel(ObjectSchema compositeType) {
                         && Objects.equals(compositeType.getLanguage().getJava().getNamespace(),
                             ClassType.POLL_OPERATION_DETAILS.getPackage())) {
                         classType = ClassType.POLL_OPERATION_DETAILS;
-                    } else if (Objects.equals(name, ClassType.REQUEST_CONDITIONS.getName())
+                    } else if (ClassType.REQUEST_CONDITIONS.getName().endsWith(name)
```

Comment:
curious, why we relaxed the check (`equals` -> `endsWith`)

Reply:
Because in client core, we named this as `HttpRequestConditions` and previously it was `RequestConditions`. So, instead of `equals`, we now have to use `endsWith` to maintain the same check. However, we should a completely different way to map TypeSpec names to class names from dependencies in the long term.